torch/nn/parallel/distributed.py

```diff
@@ -1060,22 +1060,36 @@ def to_map(obj):
             if isinstance(obj, torch.Tensor):
                 if obj.device == torch.device("cuda", target_gpu):
                     return (obj,)
+                elif hasattr(torch, "xpu") and obj.device == torch.device("xpu", target_gpu):
```

Comment:
I believe this path won't trigger if you have `device_ids = None` in the DDP ctor. device_ids are very cuda specific concept according to the [documentation](https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html#torch.nn.parallel.DistributedDataParallel).

Reply:
Hi, alanwaketan, thanks for your suggestions.
First, We have validated the case with device_ids = None, and it works in XPU.
For your question, it is normal not trigger the above path, because in the forward function, 
       def forward(self, *inputs, **kwargs):
            ......................
            if self.device_ids:
                inputs, kwargs = self.to_kwargs(inputs, kwargs, self.device_ids[0])
                output = self.module(*inputs[0], **kwargs[0])
            else:
                output = self.module(*inputs, **kwargs)

When device_ids = None, we will not call the to_kwargs functions, and naturally not trigger the above path, it is defined in _recursive_to, which is called by to_wargs.

Besides, for conditions when device_ids are given, it triggers the above path normally and XPU could work. So it also works on XPU and not cuda specific.

Reply:
Well, it's up to you. I'm not sure if `device_ids` is useful for your backend.